modules/programs/vscode.nix

```diff
@@ -34,6 +34,42 @@ let
   # TODO: On Darwin where are the extensions?
   extensionPath = ".${extensionDir}/extensions";
 
+  toExtensionJsonEntry = drv: rec {
+    identifier = {
+      id = "${drv.vscodeExtPublisher}.${drv.vscodeExtName}";
+      uuid = "";
+    };
+
+    version = drv.version;
+
+    location = {
+      "$mid" = 1;
+      fsPath = drv.outPath
+        + "/share/vscode/extensions/${drv.vscodeExtUniqueId}";
+      path = location.fsPath;
+      scheme = "file";
+    };
+
+    metadata = {
+      id = identifier.uuid;
+      publisherId = "";
+      publisherDisplayName = drv.vscodeExtPublisher;
+      targetPlatform = "undefined";
+      isApplicationScoped = false;
+      updated = false;
+      isPreReleaseVersion = false;
+      installedTimestamp = 0;
+      preRelease = false;
+    };
+  };
+
+  extensionJson = builtins.toJSON (map toExtensionJsonEntry cfg.extensions);
+  extensionJsonFile = pkgs.writeTextFile {
+    name = "extensions-json";
+    text = extensionJson;
+    destination = "/share/vscode/extensions/extensions.json";
+  };
```

Comment:
Note to self: it would be nice to deduplicate all of this between here and nixpkgs, probably by exposing it in `pkgs.vscode-utils`.

Reply:
That PR has landed in nixos-unstable, you should be able to use `text = pkgs.vscode-utils.toExtensionJson cfg.extensions;` now.